js/calc.js

```diff
@@ -0,0 +1,102 @@
+//FUNCTIONS
+function makeButtonNum(newButton, index, val){
+  newButton = numbers[index];
+  newButton.addEventListener('click',handleClick);
+  function handleClick(){
+    line += val.toString();
+    changeDisplay(line);
+  }
+  return newButton;
+}
+
+function makeOperatorButton(newButton, index, operator){
+  newButton = operators[index];
+  newButton.addEventListener('click',handleClick);
+  function handleClick(){
+    line += operator;
+    changeDisplay(line);
+  }
+}
+
+function clearLine(){
+  line = '';
+  changeDisplay(line);
+}
+
+function equalsLine(){
+  //keep a runnign total
+  var runningTotal=0;
+  //Split the line into an array whenever it reads an operator, use a regex for this
+  var linecopy = line;
+  //((/[*/+-]/) gives numbers
+  arrayNumbers = line.split(/[*/+-]/);
+  ///[^*/%+-]/) gives operators and "" at each end. pop and shift to remove
+  arrayOperators = linecopy.split(/[^*/%+-]+/);
+  arrayOperators.shift();
+  arrayOperators.pop();
+  //convert split[0] to number, set it as running total, and shift
+  runningTotal = parseInt(arrayNumbers[0]);
+  arrayNumbers.shift();
+  //get out next int ready, and shift that from numbers array
+  do{
+    var nextInt = parseInt(arrayNumbers[0]);
+    arrayNumbers.shift();
+    //use case select to find out what the operator was
+    var nextOperator = arrayOperators[0];
+    arrayOperators.shift();
+    switch (nextOperator) {
+      case '+':
+        runningTotal = runningTotal + nextInt;
+        break;
+      case '-':
+        runningTotal = runningTotal - nextInt;
+        break;
+      case '*':
+        runningTotal = runningTotal * nextInt;
+        break;
+      case '/':
+        runningTotal = runningTotal / nextInt;
+        break;
+      default:
+        alert('Invalid input!')
+        throw 'InvalidInputError'
+    }
+  }while(arrayNumbers.length>0);
+  //reset the line
+  line = ''
+  //resolves the maths
+  changeDisplay(runningTotal.toString());
+}
+
+function changeDisplay(newInput){
+  display.innerHTML = newInput;
+}
+
+//PROGRAM START
```

Comment:
A LOT of global variables here. Try to cut down on these as much as possible in the future.